packages/ember-glimmer/lib/syntax/render.js

```diff
@@ -81,7 +83,18 @@ class AbstractRenderManager {
 class SingletonRenderManager extends AbstractRenderManager {
   create(definition, args, dynamicScope) {
     let { name, env } = definition;
-    let controller = env.owner.lookup(`controller:${name}`);
+    let controller = env.owner.lookup(`controller:${name}`) || generateControllerFactory(env.owner, name);
+
+    if (dynamicScope.outletState) {
```

Comment:
While you are here – @krisselden noticed a bug: we never torn down the controller. The fix is easy, you just implement getDestructor in the `AbstractRenderManager` to destroy the controller. Can you add a test for it?